src/vcpkg/postbuildlint.cpp

```diff
@@ -1394,13 +1381,22 @@ namespace vcpkg
         if (windows_target)
         {
             Debug::println("Running windows targeting post-build checks");
+
+            auto release_lib_info = get_lib_info(fs, release_libs);
+            Optional<std::vector<Optional<LibInformation>>> debug_lib_info;
+
+            if (!build_info.policies.is_enabled(BuildPolicy::SKIP_ARCHITECTURE_CHECK) ||
```

Comment:
I'm not a fan of adding conditionals that need to be paired like this or the world ends but I don't have better ideas for now.